taxcalc/utils.py

```diff
@@ -225,38 +214,33 @@ def get_sums(df, na=False):
     pandas.Series
     """
     sums = defaultdict(lambda: 0)
-
     for col in df.columns.tolist():
         if col != 'bins':
             if na:
                 sums[col] = 'n/a'
             else:
                 sums[col] = (df[col]).sum()
-
     return pd.Series(sums, name='sums')
 
 
-def results(c):
+def results(calc):
     """
-    Gets the results from the tax calculator and organizes them into a table
+    Get results from Calculator object and organize them into a table.
 
     Parameters
     ----------
-    c : Calculator object
+    calc : Calculator object
 
     Returns
     -------
     DataFrame object
     """
+    if not hasattr(calc, 'records'):
+        msg = 'calc does not have records attribute in results(calc) utility'
+        raise ValueError(msg)
     outputs = []
     for col in STATS_COLUMNS:
-        if hasattr(c, 'records') and hasattr(c, 'policy'):
-            if hasattr(c.policy, col):
-                outputs.append(getattr(c.policy, col))
-            else:
-                outputs.append(getattr(c.records, col))
-        else:
-            outputs.append(getattr(c, col))
+        outputs.append(getattr(calc.records, col, np.nan))
```

Comment:
@talumbau said:
> This is a change in behavior. Previously, this function accepted raw DataFrames as well as Calculator objects. The idea is that the intent of this function is to collect up a certain set of attributes from the passed in object and return a collection of them. The caller of the function did not have to care if the object was a Calculator object or raw DataFrame. I would be in favor or reverting this function to its previous implementation.

Well then I don't understand why the old documentation said this:
```
Parameters
----------
c : Calculator object
```
There was nothing there saying the results function "accepted raw DataFrames as well as Calculator objects".
In fact. the old documentation was quite explicit about the results argument being a Calculator object.

If you want it to work in the undocumented manner that you describe above, then I suggest you change it to be the way you want.  You can do that as you add more tests to `test_utils.py` so that the utils code coverage is complete.